src/app/Shared/Redux/AutomatedAnalysisFilterReducer.tsx

```diff
@@ -0,0 +1,237 @@
+  automatedAnalysisUpdateCategoryIntent,
+} from './AutomatedAnalysisFilterActions';
+import { UpdateFilterOptions } from './RecordingFilterReducer';
+
+export interface AutomatedAnalysisFilterState {
+  targetFilters: TargetAutomatedAnalysisFilters[];
+  globalFilters: TargetAutomatedAnalysisGlobalFilters;
+}
+export interface TargetAutomatedAnalysisGlobalFilters {
+  filters: AutomatedAnalysisGlobalFiltersCategories;
+}
+export interface TargetAutomatedAnalysisFilters {
+  target: string; // connectURL
+  selectedCategory?: string;
+  filters: AutomatedAnalysisFiltersCategories;
+}
+
+export const createOrUpdateAutomatedAnalysisGlobalFilter = (
+  old: AutomatedAnalysisGlobalFiltersCategories,
+  { filterValue, filterKey }
+): AutomatedAnalysisGlobalFiltersCategories => {
+  const newFilters = { ...old };
+  newFilters[filterKey] = filterValue;
+  return newFilters;
+};
+
+export const createOrUpdateAutomatedAnalysisFilter = (
+  old: AutomatedAnalysisFiltersCategories,
+  { filterValue, filterKey, deleted = false, deleteOptions }: UpdateFilterOptions
+): AutomatedAnalysisFiltersCategories => {
+  let newFilterValues: any[];
+
+  if (!old[filterKey]) {
+    newFilterValues = [filterValue];
+  } else {
+    const oldFilterValues = old[filterKey] as any[];
+    if (deleted) {
+      if (deleteOptions && deleteOptions.all) {
+        newFilterValues = [];
+      } else {
+        newFilterValues = oldFilterValues.filter((val) => val !== filterValue);
+      }
+    } else {
+      newFilterValues = Array.from(new Set([...oldFilterValues, filterValue]));
+    }
+  }
+
+  const newFilters = { ...old };
+  newFilters[filterKey] = newFilterValues;
+  return newFilters;
+};
+
+export const getAutomatedAnalysisGlobalFilter = (
+  state: WritableDraft<{ globalFilters: TargetAutomatedAnalysisGlobalFilters }>
+) => {
+  return state.globalFilters;
+};
+
+export const getAutomatedAnalysisFilter = (
+  state: WritableDraft<{ targetFilters: TargetAutomatedAnalysisFilters[] }>,
+  target: string
+): TargetAutomatedAnalysisFilters => {
+  const targetFilter = state.targetFilters.filter((targetFilters) => targetFilters.target === target);
+  return targetFilter.length > 0 ? targetFilter[0] : createEmptyAutomatedAnalysisFilters(target);
+};
+
+export const createEmptyAutomatedAnalysisFilters = (target: string) =>
+  ({
+    target: target,
+    selectedCategory: 'Name',
+    filters: emptyAutomatedAnalysisFilters,
+  } as TargetAutomatedAnalysisFilters);
+
+export const deleteAllAutomatedAnalysisFilters = (automatedAnalysisFilter: TargetAutomatedAnalysisFilters) => {
+  return {
+    ...automatedAnalysisFilter,
+    selectedCategory: automatedAnalysisFilter.selectedCategory,
+    filters: emptyAutomatedAnalysisFilters,
+  };
+};
+
+// Initial states are loaded from local storage if there are any
+const initialState = {
+  state: getFromLocalStorage('AUTOMATED_ANALYSIS_FILTERS', {
+    targetFilters: [],
+    globalFilters: { filters: { Score: 0 } },
```

Comment:
I think we can reduce the nest level by removing the inner `filters`. But the logics seem to work fine so I guess we can refactor it later.

Reply:
I do remember thinking about this, but I forgot what problem I was running into. I will refactor another time I think!